### PR TITLE
#167915602 Redirect to frontend upon validation of international payment

### PR DESCRIPTION
**Pull request by patrickf949**
## Description ##
Upon validation of international payments users should be redirected to the frontend to a descriptive page informing them of the status of their transaction

## Type of change ##
Please select the relevant option
- [x] Bug fix(a non-breaking change which fixes an issue)
- [ ] New feature(a non-breaking change which adds functionality)
- [ ] Breaking change(fix of a feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ##
Please describe the tests that you ran to verify your changes
- [ ] End to end test
- [x] Integration test
- [ ] unit test

## Checklist: ##
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

## Related Pivotal Tracker stories ##
[#167915602](https://www.pivotaltracker.com/story/show/167915602)

### CODE REVIEW

**Review by bl4ck4ndbr0wn — APPROVED** [review]
LGTM

Also, try to resolve the PEP8 issues above.

File: transactions/views.py [hunk 1]
```diff
@@ -368,14 +370,14 @@ def validate_payment(request):
...
 @api_view(['GET'])
-@permission_classes((IsAuthenticated, ))
 def foreign_card_validation_response(request):
     """
     Endpoint for handling foreign card validation response. It uses the
     response to verify the payment and save the card token, if requested
     :param request: DRF request object
     :return: JSON response
     """
+    domain = os.environ.get('FRONT_END_LOGIN_URL')
```
Comment (CryceTruly) [inline transactions/views.py]: Hey @patrickf949, Just to understand here, Are we redirecting to the login URL or some other part of our frontend that is protected?

**Review by CryceTruly — CHANGES_REQUESTED** [review]

File: transactions/views.py (hunk 1 above)
Comment (patrickf949) [reply transactions/views.py]: @CryceTruly we are redirecting to the frontend Url.
The variable name will be changed

File: transactions/views.py [hunk 2]
```diff
@@ -395,11 +399,16 @@ def foreign_card_validation_response(request):
         references = {k: v for k, v in data.items() if k.endswith('ref')}
         amount = data.get('amount', 0)
         save_deposit(purpose, references, amount, user, property)
+
     else:
-        return Response({'message': verify_resp.get('message')},
-                        status=status.HTTP_400_BAD_REQUEST)
+        return HttpResponseRedirect(
+            f"{domain}/international-payment/status/" +
+            f"?message={verify_resp.get('message')}&status=fail"
```
Comment (CryceTruly) [inline transactions/views.py]: Hey @patrickf949, could we make this status as failure instead of fail?, It sounds grammatically incorrect.

**Review by CryceTruly — CHANGES_REQUESTED** [review]

File: transactions/views.py (hunk 2 above)
Comment (patrickf949) [reply transactions/views.py]: @CryceTruly done

File: transactions/views.py [hunk 3]
```diff
@@ -368,17 +371,20 @@ def validate_payment(request):
...
     response to verify the payment and save the card token, if requested
     :param request: DRF request object
     :return: JSON response
     """
+    domain = os.environ.get('FRONT_END_URL')
+
     resp = json.loads(request.query_params['response'])
     verify_resp = TransactionServices.verify_payment(resp['txRef'])
-    user = request.user
+    # user = request.user
```
Comment (Victoradukwu) [inline transactions/views.py]: @patrickf949 Let's remove the commented code

**Review by Victoradukwu — CHANGES_REQUESTED** [review]
Nice work, @patrickf949 Please check the comment I left

File: transactions/views.py (hunk 3 above)
Comment (patrickf949) [reply transactions/views.py]: Thanks @Victoradukwu, all done

**Review by Victoradukwu — APPROVED** [review]
LGTM

**Review by CryceTruly — APPROVED** [review]
LGTM

**Pull request merged**